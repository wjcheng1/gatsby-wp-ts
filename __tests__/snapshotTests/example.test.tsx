import { render } from '@testing-library/react';
import Index from '../../src/templates/index';
import { data } from '../data/allWpPost.json';

/**
 * This is an example snapshot test.
 * When the tests run, it will generate a snapshot file
 * if one is not already generated.
 * If it already exists, the component will be rendered and checked
 * against the snapshot.
 * Be sure to update your snapshots with the update-snapshots script
 * if your components markup changes.
 */
describe('<Index />', () => {
	it('should render with posts', () => {
		const { asFragment } = render(
			<Index
				pageContext={{
					posts: data.allWpPost.edges as Queries.WpPostEdge[],
				}}
			/>,
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
