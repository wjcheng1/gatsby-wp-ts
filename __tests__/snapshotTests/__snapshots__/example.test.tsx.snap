// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Index /> > should render with posts 1`] = `
<DocumentFragment>
  <div
    class="undefined flex flex-col mx-auto min-h-screen"
    data-is-root-path="true"
  >
    <nav>
      <ul
        class="undefined flex flex-row flex-wrap justify-between list-none mx-auto max-w-screen-sm"
      >
        <li>
          <a
            class="text-black no-underline"
            href="/"
          >
            üè† Home
          </a>
        </li>
        <li>
          <a
            class="text-black no-underline"
            href="/posts"
          >
            üì∞ Posts
          </a>
        </li>
        <li>
          <a
            class="text-black no-underline"
            href="/pages"
          >
            üìë Pages
          </a>
        </li>
        <li>
          <a
            class="text-black no-underline"
            href="/examples"
          >
            ‚öõÔ∏è Examples
          </a>
        </li>
      </ul>
    </nav>
    <main>
      <div>
        <h1
          class="undefined font-extrabold"
        >
          Welcome to 
          <a
            class="text-purple-600 font-medium no-underline"
            href="https://www.gatsbyjs.com/"
          >
            Gatsby!
          </a>
        </h1>
        <div>
          <span>
            Decoupled WordPress on 
          </span>
          <img
            alt="Pantheon Logo"
            height="60"
            src="/pantheon.png"
            width="191"
          />
        </div>
      </div>
      <section>
        <div>
          <a
            class="undefined rounded-lg cursor-pointer overflow-x-hidden no-underline"
            href="/posts/example-post-with-image/"
          >
            <div
              class="shrink-0 h-40 relative"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; height: 100%; width: 100%;"
              >
                <div
                  style="max-width: 640px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='360'%20width='640'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <picture>
                  <source
                    data-srcset="/static/045d1d479eed7ff0a5984b59c75d98b9/67032/pizza.webp 160w,/static/045d1d479eed7ff0a5984b59c75d98b9/c0bcc/pizza.webp 320w,/static/045d1d479eed7ff0a5984b59c75d98b9/17574/pizza.webp 640w"
                    sizes="(min-width: 640px) 640px, 100vw"
                    type="image/webp"
                  />
                  <img
                    alt="Example Post with Image"
                    data-gatsby-image-ssr=""
                    data-main-image=""
                    data-src="/static/045d1d479eed7ff0a5984b59c75d98b9/3440d/pizza.jpg"
                    data-srcset="/static/045d1d479eed7ff0a5984b59c75d98b9/e51a0/pizza.jpg 160w,/static/045d1d479eed7ff0a5984b59c75d98b9/b4dad/pizza.jpg 320w,/static/045d1d479eed7ff0a5984b59c75d98b9/3440d/pizza.jpg 640w"
                    decoding="async"
                    loading="lazy"
                    sizes="(min-width: 640px) 640px, 100vw"
                    style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; object-fit: fill; opacity: 0;"
                  />
                </picture>
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
                </script>
              </div>
            </div>
            <h2
              class="undefined font-semibold my-5 mx-4"
            >
              Example Post with Image ‚Üí
            </h2>
          </a>
          <a
            class="undefined rounded-lg cursor-pointer overflow-x-hidden no-underline"
            href="/posts/hello-world/"
          >
            <div
              class="shrink-0 h-40 relative"
            >
              <div />
            </div>
            <h2
              class="undefined font-semibold my-5 mx-4"
            >
              Hello world! ‚Üí
            </h2>
          </a>
        </div>
      </section>
    </main>
    <footer
      class="bg-black flex flex-col justify-self-end mt-12"
    >
      <nav
        class="undefined flex flex-col mx-auto"
      >
        <ul
          class="mt-4"
        >
          <li
            class="text-blue-300 list-disc ml-3"
          >
            <a
              class="text-blue-300 no-underline"
              href="/posts/example-post-with-image/"
            >
              Example Post with Image
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="text-white my-4 mx-auto p-2"
      >
        <span>
          ¬© 2023 Built with 
          <a
            class="underline"
            href="https://www.gatsbyjs.com"
          >
            Gatsby
          </a>
           and 
          <a
            class="underline"
            href="https://wordpress.org/"
          >
            WordPress
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
